src/node_nanomsg.cc

```diff
@@ -37,7 +39,11 @@ NAN_METHOD(Close) {
     rc = nn_close(s);
   } while (rc < 0 && errno == EINTR);
 
-  info.GetReturnValue().Set(Nan::New<Number>(rc));
+  v8::Local<v8::Number> ret = Nan::New<Number>(rc);
+  Local<Value> argv[] = { ret };
+
+  info.GetReturnValue().Set(ret);
+  callback->Call(1, argv);
```

Comment:
This doesn't make anything async, you're still calling the callback synchronously.  You'll need to coordinate with libuv to get notified when bind finishes.  See the work @comick did: https://github.com/comick/node-nanomsg/commit/87b2c5300b1a449b891c05fa2687690aec88731f .  Not sure if that's the correct way to interface with the event loop, but nn_bind is a blocking synchronous function, so you either have to spawn a thread and call it or work with libuv.